### DISCUSSION TITLE
Failed to start server in (production) mode when run keycloak 19 container

### DISCUSSION THREAD

**Discussion by onlookor** [keycloak/keycloak#13432] [Q&A]
As title. 
### Warns in log:

> 2022-07-30 20:03:33,924 WARN  [org.infinispan.CONFIG] (keycloak-cache-init) ISPN000569: Unable to persist Infinispan internal caches as no global state enabled
> 2022-07-30 20:03:34,027 WARN  [org.infinispan.PERSISTENCE] (keycloak-cache-init) ISPN000554: jboss-marshalling is deprecated and planned for removal
> 2022-07-30 20:03:34,459 WARN  [org.keycloak.services.DefaultComponentFactoryProviderFactory] (main) No system-wide ClusterProviderFactory found. Cannot send messages across cluster, thus disabling caching of components. Consider setting cachingForced option in single-node deployment.
> 2022-07-30 20:03:36,456 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
> 2022-07-30 20:03:36,458 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the receive buffer of socket MulticastSocket was set to 20.00MB, but the OS only allocated 212.99KB
> 2022-07-30 20:03:36,465 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
> 2022-07-30 20:03:36,471 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the receive buffer of socket MulticastSocket was set to 25.00MB, but the OS only allocated 212.99KB

### The errors and its context:

> 2022-07-30 20:03:38,539 INFO  [org.jgroups.protocols.pbcast.GMS] (keycloak-cache-init) login-13859: no members discovered after 2022 ms: creating cluster as coordinator
> 2022-07-30 20:03:38,576 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000094: Received new cluster view for channel ISPN: [login-13859|0] (1) [login-13859]
> 2022-07-30 20:03:38,598 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000079: Channel `ISPN` local address is `login-13859`, physical addresses are `[172.23.0.3:37324]`
> 2022-07-30 20:03:40,115 INFO  [org.infinispan.CLUSTER] (main) ISPN000080: Disconnecting JGroups channel `ISPN`
> 2022-07-30 20:03:40,494 **ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (production) mode**
> 2022-07-30 20:03:40,497 **ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.**

After the last error, keycloak restarted, again and again.
I have no idea about what happend.
### My dockerfile
`FROM quay.io/keycloak/keycloak:latest as builder

ENV KC_DB=postgres
ENV KC_HEALTH_ENABLED=true
ENV KC_METRICS_ENABLED=true

RUN curl -sL https://github.com/aerogear/keycloak-metrics-spi/releases/download/2.5.3/keycloak-metrics-spi-2.5.3.jar -o /opt/keycloak/providers/keycloak-metrics-spi-2.5.3.jar
RUN /opt/keycloak/bin/kc.sh build --features=admin-fine-grained-authz,docker,openshift-integration,scripts,token-exchange,map-storage,declarative-user-profile,dynamic-scopes,client-secret-rotation,recovery-codes,update-email,preview

FROM quay.io/keycloak/keycloak:latest
COPY --from=builder /opt/keycloak/ /opt/keycloak/
WORKDIR /opt/keycloak

RUN keytool -genkeypair -storepass password -storetype PKCS12 -keyalg RSA -keysize 2048 -dname "CN=login.ways.dev" -alias sso -ext "SAN:c=DNS:login.ways.dev,IP:127.0.0.1" -keystore conf/server.keystore
ENTRYPOINT ["/opt/keycloak/bin/kc.sh"]`
### My compose file:
version: '3'

name: mfasso

services:
  keycloak-db:
    build:
      dockerfile: database.dockerfile
    image: onlookor/postgres
    container_name: keycloak_postgresql
    hostname: ssodb
    restart: always
    env_file: keycloak.env

  keycloak:
    build:
      dockerfile: keycloak.dockerfile
    image: onlookor/keycloak
    hostname: login
    container_name: keycloak
    command: 
      - "start"
      - "--optimized"
      - "--db-url=jdbc:postgresql://keycloak-db:5432/keycloak"
      - "--db-username=keycloak"
      - "--db-password=test"
      - "--https-key-store-file=/opt/keycloak/conf/server.keystore"
      - "--hostname=localhost"
    restart: always
    depends_on: [keycloak-db]
    ports: ["8083:8443"]
    env_file: keycloak.env
### The keycloak.env
POSTGRES_DB=keycloak
POSTGRES_PASSWORD=test
POSTGRES_USER=keycloak
KEYCLOAK_ADMIN=admin
KEYCLOAK_ADMIN_PASSWORD=test
KC_DB=postgres

### Any suggestion would be appreciated.

**Comment by onlookor** [comment]
It's ok after I removed the line "RUN curl -sL https://github.com/aerogear/keycloak-metrics-spi……" and the feature "map-storage".
But the admin console endpoint sticks in "Loading the admin console".  After a while, the log reports "keycloak SSLEngine closed already".
what have I missed?

**Reply by bioszombie** [reply]
I have the same issue. Cannot get into the admin console at all. Just spins in an infinite loop.

**Reply by supersaiyane** [reply]
Below is my Docker-Compose file 
`version: '3.8'

networks:
  keycloak-network:
    driver: bridge

services:

  db:
    image: postgres:latest
    environment:
      - POSTGRES_DB=keycloak
      - POSTGRES_USER=keycloak
      - POSTGRES_PASSWORD=keycloak
      - PGDATA=/var/lib/postgresql/data/pgdata
    container_name: keycloak-db
    restart: unless-stopped
    ports:
      - '5432:5432'
    networks:
      - keycloak-network

  keycloak:
    container_name: keycloak
    image: quay.io/keycloak/keycloak:latest
    restart: unless-stopped
    environment:
      - KC_DB=postgres
      - KC_DB_SCHEMA=public
      - KC_DB_USERNAME=keycloak
      - KC_DB_PASSWORD=keycloak
      - KC_DB_URL_DATABASE=keycloak
      - KC_DB_URL_HOST=keycloak-db
      - KC_DB_URL_PORT=5432
      - KC_HOSTNAME=localhost:8080
      - KC_HOSTNAME_STRICT=false
      - KEYCLOAK_ADMIN=admin
      - KEYCLOAK_ADMIN_PASSWORD=admin
      - KC_HTTPS_PORT=8443
      - KC_HTTPS_PROTOCOLS=TLSv1.3,TLSv1.2
      - KC_HTTP_ENABLED="true"
      - KC_HTTP_PORT=8080
      - KC_METRICS_ENABLED=true
      - KC_HEALTH_ENABLED=true
      - KC_PROXY=edge
      - KC_HTTP_RELAITVE_PATH=/admin
      - PROXY_ADDRESS_FORWARDING=true
      - KC_FEATURES=
        authorization,
        token-exchange,
        docker,
        impersonation,
        scripts,
        upload-scripts,
        web-authn,
        client-policies,
        dynamic-scopes,
        admin-fine-grained-authz
    entrypoint: /opt/keycloak/bin/kc.sh start-dev
    ports:
      - 8080:8080
      - 8443:8443
    healthcheck:
      test:
        ["CMD-SHELL", "curl -f http://localhost:8080/ || exit 1"]
      interval: 30s
      timeout: 10s
      retries: 5  
    networks:
      - keycloak-network
    depends_on:
      - db

`

i have added start dev  in entry point  if dev , and build if prodution
